lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -612,12 +633,14 @@ def generate_private_key(
     """Generate a private key with the RSA algorithm.
 
     Args:
-        key_size (int): Key size in bytes
+        key_size (int): Key size in bits, must be at least 2048 bits
         public_exponent: Public exponent.
 
     Returns:
         PrivateKey: Private Key
     """
+    if key_size < 2048:
+        raise ValueError("Key size must be at least 2048 bits for RSA security")
     private_key = rsa.generate_private_key(
         public_exponent=public_exponent,
         key_size=key_size,
```

Comment:
From an OOP perspective, would it make more sense to have the PrivateKey class validate that it is 2048 bits, since it already has the `is_valid()` method?

Perhaps the `PrivateKey` object should own this method as a `generate()`  method. I know we can't change the signature here without breaking backwards compatibility, but it might be better just forward this on to a call like `PrivateKey.generate(key_size, public_exponent)`.

Reply:
While I agree with the suggestion here, it feels a bit out of scope for this PR. I would keep the check in the function unless we move it to the PrivateKey class. It makes the expectations from this function explicit.